src/utils/installPackage.js

```diff
@@ -3,22 +3,67 @@ const config = require('./config');
 const path = require('path');
 const promisify = require('./promisify');
 const resolve = promisify(require('resolve'));
+const commandExists = require('command-exists').sync;
+const logger = require('../Logger');
+const fs = require('./fs');
 
-async function install(dir, modules, installPeers = true) {
-  let location = await config.resolve(dir, ['yarn.lock', 'package.json']);
+async function install(
+  dir,
+  modules,
+  installPeers = true,
+  saveDev = true,
+  packageManager
+) {
+  let projectRootLocation = dir;
 
-  return new Promise((resolve, reject) => {
+  let configFileLocation = await config.resolve(dir, [
+    'yarn.lock',
+    'package.json'
+  ]);
+
+  if (configFileLocation)
+    projectRootLocation = path.dirname(configFileLocation);
+
+  return new Promise(async (resolve, reject) => {
     let install;
     let options = {
-      cwd: location ? path.dirname(location) : dir
+      cwd: projectRootLocation
     };
 
-    if (location && path.basename(location) === 'yarn.lock') {
-      install = spawn('yarn', ['add', ...modules, '--dev'], options);
+    let packageManagerToUse;
+    if (packageManager) {
+      packageManagerToUse = packageManager;
+    } else {
+      // If no package manager specified, try to figure out which one to use:
+      // Default to npm
+      packageManagerToUse = 'npm';
+      // If the yarn command exists and we find a yarn.lock, use yarn
+      if (commandExists('yarn')) {
```

Comment:
`if (commandExists('yarn') && await fs.exists(...))`

Reply:
There’s a reason I separated them. I only want the `else` to apply if `await fs.exists(...)` is **false** **AND** `commandExists(‘yarn’)` is **true**.

Pretty much only if the user has yarn, but no `yarn.lock`, then I want to trigger the warning. If they don’t have yarn, then no need to warn them.

Reply:
Ah I see. The warning may be unnecessary. If they haven't explicitly asked for yarn, and there is no yarn.lock, then using npm without a warning seems valid - expected even.

Reply:
Pretty much the same edge case that I mentioned before applies. If you’re starting a new project, you’d want to know that you need to create a `yarn.lock` / `--yarn` and that Parcel isn’t using yarn.

I see your point though... if you want, I can remove it